### PR TITLE
[TGUI] Robotics Control Console

### PR DESCRIPTION
**Pull request by Kyep**
## What Does This PR Do
Converts robotics console to TGUI.

## Images of changes
Standard view (RD/Robotocist):
TGUI_robotics_console
Malf AI view, showing hacked/hackable borgs:
TGUI-robotics-AI-malf-view

## Changelog
:cl: Kyep
tweak: Converted robotics console to TGUI.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/robot.dm [hunk 1]
```diff
@@ -20,221 +20,179 @@
...
-	return 0
-
-/obj/machinery/computer/robotics/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+		return TRUE
+	if(allowed(user))
+		return TRUE
+	return FALSE
+
+/// Does this robot show up on the console?
```
Comment (AffectedArc07) [inline code/game/machinery/computer/robot.dm:34]:
```suggestion
/**
  * Does this robot show up in the console
  *
  * Returns TRUE if a robot will show up in the console
  * Returns FALSE if a robot will not show up in the console
  * Arguments:
  * * R - The [mob/living/silicon/robot] to be checked
  */
```

Procs can be documented in a much cleaner way like this
```

File: code/game/machinery/computer/robot.dm [hunk 2]
```diff
@@ -20,221 +20,179 @@
...
+				to_chat(user, "<span class='warning'>AIs can only control cyborgs which are linked to them.</span>")
+			return FALSE
+	if(isrobot(user))
+		if(R != user)
+			if(telluserwhy)
+				to_chat(user, "<span class='warning'>Cyborgs cannot control other cyborgs.</span>")
+			return FALSE
+	return TRUE
+
+/// Display hacking options when viewing console?
```
Comment (AffectedArc07) [inline code/game/machinery/computer/robot.dm:64]:
```suggestion
/**
  * Check if the user is allowed to hack any cyborgs
  *
  * Returns TRUE if a user can hack any cyborg
  * Returns FALSE if a user cannot
  * Arguments:
  * * user - The [mob/user] to be checked
  */
```

Procs can be documented in a much cleaner way like this

File: code/game/machinery/computer/robot.dm [hunk 3]
```diff
@@ -20,221 +20,179 @@
...
+/obj/machinery/computer/robotics/proc/can_hack_any(mob/user)
+	if(!istype(user))
+		return FALSE
+	if(user.can_admin_interact())
+		return TRUE
+	if(!isAI(user))
+		return FALSE
+	return (user.mind.special_role && user.mind.original == user)
+
+/// Can user hack this specific borg?
```
Comment (AffectedArc07) [inline code/game/machinery/computer/robot.dm:74]:
```suggestion
/**
  * Check if the user is allowed to hack a specific borg
  *
  * Returns TRUE if a user can hack the specific cyborg
  * Returns FALSE if a user cannot
  * Arguments:
  * * user - The [mob/user] to be checked
  * * R - The [mob/living/silicon/robot] to be checked
  */
```

Procs can be documented in a much cleaner way like this

File: code/game/machinery/computer/robot.dm [hunk 4]
```diff
@@ -20,221 +20,179 @@
...
+			if(!can_control(usr, R, TRUE))
+				return
+			message_admins("<span class='notice'>[ADMIN_LOOKUPFLW(usr)] [!R.lockcharge ? "locked down" : "released"] [ADMIN_LOOKUPFLW(R)]!</span>")
+			log_game("[key_name(usr)] [!R.lockcharge ? "locked down" : "released"] [key_name(R)]!")
+			R.SetLockdown(!R.lockcharge)
+			to_chat(R, "[!R.lockcharge ? "<span class='notice'>Your lockdown has been lifted!" : "<span class='alert'>You have been locked down!"]</span>")
+			if(R.connected_ai)
+				to_chat(R.connected_ai, "[!R.lockcharge ? "<span class='notice'>NOTICE - Cyborg lockdown lifted</span>" : "<span class='alert'>ALERT - Cyborg lockdown detected</span>"]: <a href='?src=[R.connected_ai.UID()];track=[html_encode(R.name)]'>[R.name]</a></span><br>")
+			. = TRUE
+		if("magbot") // AIs hacking/emagging a borg
```
Comment (AffectedArc07) [inline code/game/machinery/computer/robot.dm:187]: `magbot`? Is this a typo from `emagbot`?

File: tgui/packages/tgui/interfaces/RoboticsControlConsole.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        key={cyborg.uid}
+        title={cyborg.name}
+        buttons={(
+          <Fragment>
+            {!!cyborg.hackable && !cyborg.emagged && (
+              <Button
+                icon="terminal"
+                content="Hack"
+                color="bad"
+                onClick={() => act('magbot', {
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/RoboticsControlConsole.js:60]: Sam ehre

File: code/game/machinery/computer/robot.dm (hunk 4 above)
Comment (Kyep) [reply code/game/machinery/computer/robot.dm]: TG calls it "magbot", I assume as an abbreviation.
I'm just renaming it to "hackbot" as that is clearer for what it actually does.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**